Tests/Unit/Core/ResponseHandlerTest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+use OxidEsales\Eshop\Application\Model\Order;
+
+use Wirecard\Oxid\Core\ResponseHandler;
+
+use Wirecard\PaymentSdk\BackendService;
+use Wirecard\PaymentSdk\Entity\Amount;
+use Wirecard\PaymentSdk\Response\SuccessResponse;
+
+class ResponseHandlerTest extends \Wirecard\Test\WdUnitTestCase
+{
+
+    public function testOnSuccessResponse()
+    {
+
+        $oResponseStub = $this->getMockBuilder(SuccessResponse::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $oResponseStub->method('getParentTransactionId')
+            ->willReturn('');
+
+        $oResponseStub->method('getRequestedAmount')
+            ->willReturn(new Amount(30, "EUR"));
+
+        $oBackendService = $this->getMockBuilder(BackendService::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $oOrder = $this->getMockBuilder(Order::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        try {
+            ResponseHandler::onSuccessResponse($oResponseStub, $oBackendService, $oOrder);
```

Comment:
The only good way of testing this is looking into the db if the transaction is saved correctly. Else this test is just for coverage.

Reply:
👍